### PR TITLE
woo-fiat-premium

### PR DESCRIPTION
**Pull request by ntheile**
Screen Shot 2023-12-14 at 12 10 35 AM

### CODE REVIEW

File: plugin/zaprite.php [hunk 1]
```diff
@@ -289,23 +279,52 @@ function zaprite_server_add_update_status_callback($data)
...
+            }
+
+            // check status
+            $status = $orderStatusRes['response']['status']; // processing (aka paid), underpaid, overpaid or btc-pending
+            error_log("ZAPRITE: order status update from zaprite api - $status");
+            $wooStatus = Utils::convert_zaprite_order_status_to_woo_status($status);
+            error_log("ZAPRITE: wooStatus - $wooStatus");
+
+            switch ($wooStatus) {
+                case '':
```
Comment (testerez) [inline plugin/zaprite.php:303]: I think this should be the `case default:` once we covered all supported status

File: plugin/zaprite.php [hunk 2]
```diff
@@ -208,7 +198,7 @@ public function process_payment($order_id)
 
                 // save zaprite metadata in woocommerce
                 $order->add_meta_data('zaprite_order_id', $order_id, true);
-                $order->add_meta_data('zaprite_order_link', "$zaprite_url/_domains/pay/order/$order_id", true);
+                $order->add_meta_data('zaprite_order_link', "$zaprite_url/org/default/orders/$order_id", true);
```
[the remark this answers is not in the document]
Comment (testerez) [reply plugin/zaprite.php:201]: Here you go: https://github.com/ZapriteApp/zaprite-v2/pull/1144

File: plugin/zaprite.php [hunk 3]
```diff
@@ -289,23 +279,52 @@ function zaprite_server_add_update_status_callback($data)
...
                     return new WP_REST_Response('Invalid order status.', 400);
-                }
-                if($wooStatus == "processing") {
+                    break;
+                case 'processing':
                     $order->add_order_note('Payment is settled.');
+                    // check if fiat premium was applied, if so, save to custom data in woo
+                    $paidPremium = $data->get_param('paidPremium');
+                    $paidPremiumCurrency = $data->get_param('paidPremiumCurrency');
+                    $paidPremiumPercent = $data->get_param('paidPremiumPercent');
```
[the remark this answers is not in the document]
Comment (ntheile) [reply plugin/zaprite.php]: fixed here https://github.com/ZapriteApp/zaprite-v2/pull/1143/commits/405f6fb75c1cca5190cada70ae11bed8903f6994

File: plugin/zaprite.php [hunk 4]
```diff
@@ -289,23 +279,52 @@ function zaprite_server_add_update_status_callback($data)
...
+            if ($key !== $keyToCheck) {
+                return new WP_Error(
+                    'unauthorized_access',
+                    'Unauthorized access - keys do not match',
+                    array( 'status' => 401 )
+                );
+            }
+
+            // check status
+            $status = $orderStatusRes['response']['status']; // processing (aka paid), underpaid, overpaid or btc-pending
```
[the remark this answers is not in the document]
Comment (ntheile) [reply plugin/zaprite.php:297]: yep, just a quick note on mapping from zaprite status to woo status. i can remove now i think it wrote it before i had the util

[the remark this answers is not in the document]
Comment (ntheile) [reply plugin/zaprite.php]: i cleaned up the order status mappings to make it more clear here https://github.com/ZapriteApp/zaprite-for-woocommerce/pull/8/commits/5e56071a3d7a34be86d09caf30c9fcff3d9ffaac

i also copied your refactor into this PR from over here https://github.com/ZapriteApp/zaprite-for-woocommerce/pull/9

**Review by testerez — COMMENTED** [review]
Looks good but you may have to adapt. See my comments here: https://github.com/ZapriteApp/zaprite-v2/commit/32fedcd5aedcf84bcaa246c92c5b5cfe358efcdc#diff-895b7d689d65f77bc98a05379127f9256ee4b897aa0e0d19dfb2da38df11421aR215-R228

**Pull request merged**